Change user role triggered reset password issue resolved

TahaKhanAbdalli opened the pull request:
fixes #979

IanMayo commented:
Thanks @TahaKhanAbdalli - could you through the strategy for this fixes the issue please?

TahaKhanAbdalli commented:
While editing roles, we mistakenly triggered a password reset by including an empty string in the PUT request payload. To fix this, I removed the empty string password from the payload, avoiding unintended password resets. 
Additionally, I implemented session storage to enable users to log in again if they close the browser directly. Session storage stores data temporarily during a user's session, but it gets cleared when the user closes the browser tab or window, ensuring privacy and security.

IanMayo commented on src/App.tsx:
```diff
@@ -232,6 +232,12 @@ function App(): React.ReactElement {
   useEffect(() => {
+    const storedSessionData = sessionStorage.getItem('login')
+    if (storedSessionData !== null) {
+      console.log('Stored Value:', storedSessionData)
```
I don't think we need to log this message, once we know the code is working.

IanMayo commented:
Could you add a comment near where you set the token, to explain what is happening please?

IanMayo commented on src/App.tsx:
```diff
@@ -232,6 +232,12 @@ function App(): React.ReactElement {
   }
 
   useEffect(() => {
+    const storedSessionData = sessionStorage.getItem('login')
```
we should use a constant instead of `login`, please.  Where we define the constant, we should explain what it is being used for.

IanMayo commented on src/App.tsx:
```diff
@@ -232,6 +232,12 @@ function App(): React.ReactElement {
+    if (storedSessionData !== null) {
+      console.log('Stored Value:', storedSessionData)
+    } else {
+      removeToken()
```
@TahaKhanAbdalli - could you please rename `removeToken()` to `removeUserToken()` - since it is removing a specific token.

TahaKhanAbdalli replied:
Sure.
